src/pb_utils.h

```diff
@@ -255,7 +259,27 @@ class CUDAHandler {
       int64_t memory_type_id, cudaIpcMemHandle_t* cuda_mem_handle,
       void** data_ptr);
   void CloseCudaHandle(int64_t memory_type_id, void* data_ptr);
+
+  /// Set the device only if the primary context has already been created for
+  /// this device. Inspired from PyTorch's MaybeSetDevice.
+  /// \param device The cuda device index.
+  void MaybeSetDevice(int device);
+};
+
+
+/// A helper class to change the current device and restore the old context. The
+/// old context will be restored only when if the primary context for that
```

Comment:
*be restored only ~when~ if the primary ..

Reply:
good catch. Fixed.